src/autocat/learning/featurizers.py

```diff
@@ -109,10 +114,79 @@ def copy(self):
             design_space_structures=ds_structs_copy,
             species_list=self.species_list.copy(),
             max_size=self.max_size,
+            preset=self.preset,
             kwargs=copy.deepcopy(self.kwargs) if self.kwargs else None,
         )
         return feat
 
+    def to_jsonified_dict(self) -> Dict:
+        # collect design space structures
+        if self.design_space_structures is not None:
+            collected_structs = []
+            for struct in self.design_space_structures:
+                collected_structs.append(atoms_encoder(struct))
+        else:
+            collected_structs = None
+        mod_string = self.featurizer_class.__module__
+        class_string = self.featurizer_class.__name__
+        return {
+            "design_space_structures": collected_structs,
+            "species_list": self.species_list if self.species_list else None,
+            "preset": self.preset if self.preset else None,
+            "kwargs": self.kwargs if self.kwargs else None,
+            "max_size": self.max_size if self.max_size else None,
+            "featurizer_class": [mod_string, class_string],
+        }
+
+    def write_json_to_disk(self, write_location: str = ".", json_name: str = None):
+        """
+        Writes `Featurizer` to disk as a json
+        """
+        jsonified_list = self.to_jsonified_dict()
+
+        if json_name is None:
+            json_name = "featurizer.json"
+
+        json_path = os.path.join(write_location, json_name)
+
+        with open(json_path, "w") as f:
+            json.dump(jsonified_list, f)
+
+    @staticmethod
+    def from_jsonified_dict(all_data: Dict):
+        extracted_dict = {}
+        for prop in [
+            "species_list",
+            "preset",
+            "kwargs",
+            "max_size",
+            "design_space_structures",
+        ]:
+            extracted_dict[prop] = all_data.get(prop)
```

Comment:
I think the `extracted_dict` is superfluous. We can use `all_data.get` everywhere instead of constructing the additional dictionary.